lib/helpers.js

```diff
@@ -41,70 +41,28 @@ exports.copyLocalZip = function (localZipPath, cb) {
 
 exports.unzipFile = function (zipPath, cb) {
   logger.debug("Unzipping " + zipPath);
-  exports.testZipArchive(zipPath, function (err, valid) {
-    if (valid) {
-      if (exports.isWindows()) {
-        var zip = new AdmZip(zipPath);
-        zip.extractAllTo(path.dirname(zipPath), true);
-        logger.debug("Unzip successful");
-        cb(null, null);
-      } else {
-        var execEnv = _.clone(process.env);
-        delete execEnv.UNZIP;
-        var execOpts = {cwd: path.dirname(zipPath), maxBuffer: 524288,
-                        env: execEnv};
-        exec('unzip -o ' + zipPath, execOpts, function (err, stderr, stdout) {
-          if (!err) {
-            logger.debug("Unzip successful");
-            cb(null, stderr);
-          } else {
-            logger.error("Unzip threw error " + err);
-            logger.error("Stderr: " + stderr);
-            logger.error("Stdout: " + stdout);
-            cb("Archive could not be unzipped, check appium logs.", null);
-          }
-        });
-      }
-    } else {
-      cb(err, null);
-    }
-  });
-};
-
-exports.testZipArchive = function (zipPath, cb) {
-  logger.debug("Testing zip archive: " + zipPath);
-  if (exports.isWindows()) {
-    if (fs.existsSync(zipPath)) {
-      logger.debug("Zip archive tested clean");
-      cb(null, true);
+    if (exports.isWindows()) {
+      var zip = new AdmZip(zipPath);
+      zip.extractAllTo(path.dirname(zipPath), true);
+      logger.debug("Unzip successful");
+      cb(null, null);
     } else {
-      cb("Zip archive was not found.", false);
-    }
-  } else {
-    var execEnv = _.clone(process.env);
-    delete execEnv.UNZIP;
-    var execOpts = {cwd: path.dirname(zipPath), maxBuffer: 524288,
-                    env: execEnv};
-    exec("unzip -tq " + zipPath, execOpts, function (err, stderr, stdout) {
-      if (!err) {
-        if (/No errors detected/.exec(stderr)) {
-          logger.debug("Zip archive tested clean");
-          cb(null, true);
+      var execEnv = _.clone(process.env);
+      delete execEnv.UNZIP;
+      var execOpts = {cwd: path.dirname(zipPath), maxBuffer: 524288,
+                      env: execEnv};
+      exec('unzip -o ' + zipPath, execOpts, function (err, stderr, stdout) {
+        if (!err) {
+          logger.debug("Unzip successful");
+          cb(null, stderr);
         } else {
-          logger.error("Zip file " + zipPath + " was not valid");
+          logger.error("Unzip threw error " + err);
           logger.error("Stderr: " + stderr);
           logger.error("Stdout: " + stdout);
-          cb("Zip archive did not test successfully, check appium server logs " +
-             "for output", false);
+          cb("Archive could not be unzipped, check appium logs.", null);
```

Comment:
wait, why is all this code duplicated between appium and appium-adb? it should definitely not be.